geoapps_utils/run.py

```diff
@@ -80,6 +92,226 @@ def fetch_driver_class(json_dict: str | Path | dict) -> type[Driver]:
+        raise RuntimeError("Failed to copy the UIJsonGroup.")
+
+    if monitoring_directory is not None:
+        new_out_group.modify_option("monitoring_directory", str(monitoring_directory))
+
+    return new_out_group, new_workspace
+
+
+def run_from_outgroup_name(
+    workspace_path: str | Path,
+    name: str,
+    *,
+    destination: Path | str | None = None,
+    new_workspace_name: str | None = None,
+    monitoring_directory: Path | str | None = None,
+) -> Driver:
+    """
+    Run the function defined in the 'out_group' UIJsonGroup.
+
+    :param workspace_path: Path to a geoh5 file.
+    :param name: Name or str UUID of a UIJsonGroup with options set.
+    :param destination: Path to copy the ui.json file to. If None, a temporary directory is used.
+    :param new_workspace_name: New geoh5 file name. If None, the original name is kept.
+    :param monitoring_directory: New monitoring directory. If None, the original is kept.
+
+    :return: Driver instance.
+    """
+    if not Path(workspace_path).is_file():
+        raise FileNotFoundError(f"Workspace file '{workspace_path}' does not exist.")
+
+    with Workspace(workspace_path) as workspace:
+        out_group = workspace.get_entity(name)[0]
+        if not isinstance(out_group, UIJsonGroup):
+            raise TypeError(
+                f"Entity with name or id '{name}' "
+                f"is not a UIJsonGroup. Got {type(out_group)}."
+            )
+
+        new_workspace = None
+        if destination is not None:
+            out_group, new_workspace = copy_out_group(
+                out_group,
+                destination,
+                new_workspace_name=new_workspace_name,
+                monitoring_directory=monitoring_directory,
+            )
+
+        driver_instance = run_uijson_group(out_group)
+
+        if new_workspace is not None:
+            new_workspace.close()
+
+    return driver_instance
+
+
+def run_from_uijson(
+    uijson_path: str | Path,
+    *,
+    destination: Path | str | None = None,
+    new_workspace_name: str | None = None,
+    monitoring_directory: Path | str | None = None,
+) -> Driver:
+    """
+    Run a ui.json file, optionally copying it to a new location and changing the geoh5 file name.
+
+    :param uijson_path: Path to a ui.json file.
+    :param destination: Path to copy the ui.json file to. If None, a temporary directory is used.
+    :param new_workspace_name: New geoh5 file name. If None, the original name is kept.
+    :param monitoring_directory: New monitoring directory. If None, the original is kept.
+
+    :return: Driver instance.
+    """
+    if destination is not None:
+        uijson_path = copy_uijson_file(
+            uijson_path,
+            destination,
+            new_workspace_name=new_workspace_name,
+            monitoring_directory=monitoring_directory,
+        )
+
+    driver_instance = run_uijson_file(uijson_path)
+
+    return driver_instance
```

Comment:
After out conversation, it would be great if we could have the option to do shutils.copy instead of loading the file as the current `copy_uijson_file` does. This is to be able to eventually replace the mechanics added here: 
https://github.com/MiraGeoscience/simpeg-drivers/blob/95ad8de48f7ddd86a99b3d4b467ea566f12a2a60/simpeg_drivers/plate_simulation/sweep/driver.py#L112

Currently it makes the copy from the geoh5 and looks for the UIJson group inside, but could just do the same shutils.copy based on a template ui.json file instead.